HellBrick.NoCapture.Test/NoCaptureTest.cs

```diff
@@ -288,5 +288,195 @@ class C
 "
 			)
 			.ShouldHaveDiagnostic( "Invoke", "transforms", "this" );
+
+
+		[Fact]
+		public void NonCapturingArgumentPassedAsNoCaptureButContainingNestedCapturingLambdaIsIgnored()
+			=> _verifier
+			.Source
+			(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+[AttributeUsage( AttributeTargets.Parameter | AttributeTargets.Method )]
+class NoCaptureAttribute : Attribute
+{
+}
+
+class C
+{
+	public void CallSite() => Invoke( ( list, v ) => list.Select( x => x + v ) );
+	public void Invoke( [NoCapture] Expression<Func<List<int>, int, IEnumerable<int>>> f ) { }
+}
+"
+			)
+			.ShouldHaveNoDiagnostics();
+
+		[Fact]
+		public void CapturingArgumentPassedAsNoCaptureButContainingNestedCapturingLambdaIsReported()
+			=> _verifier
+			.Source
+			(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+[AttributeUsage( AttributeTargets.Parameter | AttributeTargets.Method )]
+class NoCaptureAttribute : Attribute
+{
+}
+
+class C
+{
+	private readonly int _x = 42;
+	public void CallSite() => Invoke( ( list, v ) => list.Aggregate( _x, ( agg, x ) => agg + x + v ) );
+	public void Invoke( [NoCapture] Expression<Func<List<int>, int, int>> f ) { }
+}
+"
+			)
+			.ShouldHaveDiagnostic( "Invoke", "f", "this" );
+
+		[Fact]
+		public void CapturingArgumentPassedAsNoCaptureButContainingNestedCapturingLambdaWhichCapturesTheSameThingCapturedInItsParentIsReported()
+			=> _verifier
+			.Source
+			(
+@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+[AttributeUsage( AttributeTargets.Parameter | AttributeTargets.Method )]
+class NoCaptureAttribute : Attribute
+{
+}
+
+class C
+{
+	private readonly int _x = 42;
+	public void CallSite() => Invoke( ( list, v ) => list.Aggregate( _x, ( agg, x ) => _x + agg + x + v ) );
+	public void Invoke( [NoCapture] Expression<Func<List<int>, int, int>> f ) { }
+}
+"
+			)
+			.ShouldHaveDiagnostic( "Invoke", "f", "this" );
+
+		[Fact]
+		public void NestedCapturingFromTheOuterArgumentIsIgnored()
+			=> _verifier
+			.Source
+			(
+@"
+using System;
+using System.Collections.Generic;
+
+static class InvokeExtension
+{
+	public static int Invoke( this List<int> list, Func<int, int> inner ) => inner( 42 );
+}
+
+class C
+{
+	private readonly int _x = 42;
+	public void CallSite() => InvokeOuter( ( list, v ) => list.Invoke( _ => v ) );
```

Comment:
Shouldn't it be `list.Invoke( _ => _x )` or `list.Invoke( _ => _x + v )`? Otherwise, despite the test name, the call site doesn't seem to be capturing anything.

Reply:
`Invoke`'s parameter lambda is capturing `InokeOuter` parameter lambda's parameter, which is the original #6 case

Reply:
Oh, right, got it =)